### PR TITLE
Fix bug #17403 by double check collmeta prefix and alias prefix

### LINKED ISSUE #17403: [Bug]: Alter alias not working after restarting docker compose, collection failed  to load
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.0.2
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.0.2
- OS(Ubuntu or CentOS): Ubuntu 20.04
- CPU/Memory: Intel Core i7-11700k / 16gb
- GPU: -
- Others: -
```

### Current Behavior

After I shutdown the service using docker compose down, and reinitialize the service again using the docker compose up. Failed to alter alias and some collection ( not all ) failed to load. 

As a context as milvus doesn't have list alias API yet, I try to embed the collection name into it's description, so to know which alias map to whic collection, I just describe the alias collection and read the description to find the collection it's mapped to. Something like this ( the alias is `propertyUnits` and the literal collection name is `propertyUnits__v0_1`

```
<Collection>:
-------------
<name>: propertyUnits
<partitions>: [{"name": "_default", "collection_name": "propertyUnits", "description": ""}]
<description>: propertyUnits__v0_1
...
}
``` 

I found that after I downed and restarted the docker compose, alter alias not working directly. The alias only changed if I restarted the service again and the cycle continue. 

In addition I also encounter a collection load error ( only 1 apart from 4 collection I have ):

```
2022-06-07 07:14:21,155 : ERROR : RPC error: [wait_for_loading_collection], <BaseException: (code=1, message=collection 433716601692094465 has not been loaded to memory or load failed)>, <Time:{'RPC start': '2022-06-07 07:14:20.651431', 'RPC error': '2022-06-07 07:14:21.155698'}>
2022-06-07 07:14:21,155 : ERROR : RPC error: [load_collection], <BaseException: (code=1, message=collection 433716601692094465 has not been loaded to memory or load failed)>, <Time:{'RPC start': '2022-06-07 07:14:20.635769', 'RPC error': '2022-06-07 07:14:21.155832'}>
*** pymilvus.client.exceptions.BaseException: <BaseException: (code=1, message=collection 433716601692094465 has not been loaded to memory or load failed)>
``` 
The order of sequence I do :

1. docker compose up
2. create collections `propertyUnits__v0_1`
3. create collections `propertyUnits__v0_2`
4. insert data to both collections
5. create alias `propertyUnits` and assign to `propertyUnits__v0_1`
6. describe the `propertyUnits` alias ( working properly, mapped to `propertyUnits__v0_1` )
7. alter alias `propertyUnits` to `propertyUnits__v0_2`
8. describe the `propertyUnits` alias ( working properly, mapped to `propertyUnits__v0_2` )
9. docker compose down
10. docker compose up
11. alter alias `propertyUnits` to `propertyUnits__v0_1`
12. describe the `propertyUnits` alias ( not working, still mapped to `propertyUnits__v0_2` )
13. docker compose down
14. docker compose up
15. describe the `propertyUnits` alias ( now mapped to `propertyUnits__v0_1` )
16. alter alias `propertyUnits` to `propertyUnits__v0_2`
17. describe the `propertyUnits` alias ( not working, still mapped to `propertyUnits__v0_1` )
18. docker compose down
19. docker compose up
20. describe the `propertyUnits` alias ( now mapped to `propertyUnits__v0_2`)

### Expected Behavior

- Alter alias working immediately even after service restart
- No collection load error

### Steps To Reproduce

```markdown
The order of sequence I do :

1. docker compose up
2. create collections `propertyUnits__v0_1`
3. create collections `propertyUnits__v0_2`
4. insert data to both collections
5. create alias `propertyUnits` and assign to `propertyUnits__v0_1`
6. describe the `propertyUnits` alias ( working properly, mapped to `propertyUnits__v0_1` )
7. alter alias `propertyUnits` to `propertyUnits__v0_2`
8. describe the `propertyUnits` alias ( working properly, mapped to `propertyUnits__v0_2` )
9. docker compose down
10. docker compose up
11. alter alias `propertyUnits` to `propertyUnits__v0_1`
12. describe the `propertyUnits` alias ( not working, still mapped to `propertyUnits__v0_2` )
13. docker compose down
14. docker compose up
15. describe the `propertyUnits` alias ( now mapped to `propertyUnits__v0_1` )
16. alter alias `propertyUnits` to `propertyUnits__v0_2`
17. describe the `propertyUnits` alias ( not working, still mapped to `propertyUnits__v0_1` )
18. docker compose down
19. docker compose up
20. describe the `propertyUnits` alias ( now mapped to `propertyUnits__v0_2`)
```

### Milvus Log

No error on alter alias

Error on collection load : 

```
2022-06-07 07:14:21,155 : ERROR : RPC error: [wait_for_loading_collection], <BaseException: (code=1, message=collection 433716601692094465 has not been loaded to memory or load failed)>, <Time:{'RPC start': '2022-06-07 07:14:20.651431', 'RPC error': '2022-06-07 07:14:21.155698'}>
2022-06-07 07:14:21,155 : ERROR : RPC error: [load_collection], <BaseException: (code=1, message=collection 433716601692094465 has not been loaded to memory or load failed)>, <Time:{'RPC start': '2022-06-07 07:14:20.635769', 'RPC error': '2022-06-07 07:14:21.155832'}>
*** pymilvus.client.exceptions.BaseException: <BaseException: (code=1, message=collection 433716601692094465 has not been loaded to memory or load failed)>
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by EricStarer**
Signed-off-by: xingzhao <xing.zhao@zilliz.com>

Fix https://github.com/milvus-io/milvus/issues/17403

### CODE REVIEW

File: internal/rootcoord/meta_table.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ const (
 	IndexMetaPrefix = ComponentPrefix + "/index"
 
 	// CollectionAliasMetaPrefix prefix for collection alias meta
-	CollectionAliasMetaPrefix = ComponentPrefix + "/collection-alias"
+	CollectionAliasMetaPrefix = ComponentPrefix + "/alias"
```
Comment (congqixia) [inline internal/rootcoord/meta_table.go:57]: Will there be any issue if any old version milvus upgrade to latest wit this prefix changed?

Comment (EricStarer) [reply internal/rootcoord/meta_table.go:57]: Yes, old alias meta will not be able to load. According to the issue, the alias function is faulty now，so I guess few clients are using alias function.

**Comment by xiaofan-luan** [comment]
1. one way to fix that is to double check if it's ComponentPrefix + /collection  or ComponentPrefix + /collection-alias when load.
2. add some unit test on it to verify it works

**Comment by EricStarer** [comment]
> 1. one way to fix that is to double check if it's ComponentPrefix + /collection  or ComponentPrefix + /collection-alias when load.
> 2. add some unit test on it to verify it works

OK

File: internal/rootcoord/meta_table.go [hunk 2]
```diff
@@ -151,6 +151,21 @@ func (mt *MetaTable) reloadFromKV() error {
...
+	if err != nil {
+		return err
+	}
+	aliases := make([]string, 0, len(values))
+	for _, value := range values {
+		aliasInfo := pb.CollectionInfo{}
+		err = proto.Unmarshal([]byte(value), &aliasInfo)
+		if err != nil {
+			return fmt.Errorf("rootcoord Unmarshal pb.AliasInfo err:%w", err)
+		}
```
Comment (congqixia) [inline internal/rootcoord/meta_table.go:164]: The error handling is tricky here. If some etcd meta is corrupted, `RootCoord` will not provide service until some manual interference is involved.

Comment (EricStarer) [reply internal/rootcoord/meta_table.go:164]: So what‘s the correct behavior when some etcd meta is corrupted?

Comment (EricStarer) [reply internal/rootcoord/meta_table.go:164]: I see in the unified catlog pr, the error becomes the warn log. Maybe we can first merge this pr and then do some fix.

File: internal/rootcoord/meta_table.go [hunk 3]
```diff
@@ -162,6 +177,16 @@ func (mt *MetaTable) reloadFromKV() error {
 		if err != nil {
 			return fmt.Errorf("rootcoord Unmarshal pb.CollectionInfo err:%w", err)
 		}
+		flag := false
```
Comment (xiaofan-luan) [inline internal/rootcoord/meta_table.go:180]: why not use mt.collAlias2ID as the check?

**Comment by jennyli-z** [comment]
image
If this pr will fix the issue #17403 , you can add `fix` before the issue so that the issue can be automatically closed after this pr is merged.
like https://github.com/milvus-io/milvus/pull/17547
image

**Pull request merged**